### DISCUSSION TITLE
How to obtain the Effective Sample Size per gradient evaluation in BlackJax?

### DISCUSSION THREAD

**Discussion by Madhav-Kanda** [blackjax-devs/blackjax#559] [Q&A]
I wish to calculate the effective sample size (ESS) per gradient evaluation. How can I obtain it?

**Comment by junpenglao** [comment]
🤖 *This reply was drafted with Claude and reviewed by me (@junpenglao).*

ESS-per-gradient is ESS divided by the total number of gradient evaluations. For NUTS, sum `info.num_integration_steps` over the chain (each integration step ≈ one gradient eval with velocity-Verlet), then:

```python
ess = blackjax.ess(samples)  # blackjax.diagnostics.effective_sample_size
ess_per_grad = ess / info.num_integration_steps.sum()
```
